pyEpiabm/pyEpiabm/sweep/travel_sweep.py

```diff
@@ -0,0 +1,251 @@
+                    microcells_to_choose_dict[possible_microcell] = \
+                        len(possible_microcell.persons)
+
+        # Assign to microcell and household in existing population
+        for person in self.initial_cell.persons:
+            selected_microcell = random.choice(list(
+                microcells_to_choose_dict.keys()))
+            # Assign person to microcell and microcell to person
+            selected_microcell.add_person(person)
+            person.microcell = selected_microcell
+            r = random.random()
+            if r < self.travel_params['prob_existing_household']:
+                # Assign to existing household
+                selected_household = random.choice(
+                    selected_microcell.households)
+                selected_household.add_person(person)
+            else:
+                # Create new household
+                selected_microcell.add_household([person])
+
+    def check_leaving_individuals(self, time, person):
+        """
+        Check if individuals travel_end_time is reached. If interventions
+        are active and travelling individual is in isolation and/or quarantine
+        individuals leave after their travel_end_time is reached and they are
+        out of isolation and/or quarantine.
+
+        Parameters
+        ----------
+        time : float
+            Simulation time
+        Person : Person
+            Instance of Person class
+
+        """
+        remove_individual = False
+        if (hasattr(person, 'travel_end_time')) and \
+                (time > person.travel_end_time):
+            if all([hasattr(person, 'isolation_start_time'),
+                    hasattr(person, 'quarantine_start_time')]):
+                if (person.isolation_start_time and
+                        person.quarantine_start_time is None) or \
+                        (person.isolation_start_time and
+                         person.quarantine_start_time <= time):
+                    remove_individual = True
+            elif hasattr(person, 'isolation_start_time'):
+                if (person.isolation_start_time is None) or \
+                        (person.isolation_start_time <= time):
+                    remove_individual = True
+            elif hasattr(person, 'quarantine_start_time'):
+                if (person.quarantine_start_time is None) or \
+                        (person.quarantine_start_time <= time):
+                    remove_individual = True
+            else:
+                remove_individual = True
+        return remove_individual
+
+    def remove_leaving_individuals(self, time):
+        """
+        Remove individuals after their travel_end_time is reached
+        from cell, microcell and household.
+
+        Parameters
+        ----------
+        time : float
+            Simulation time
+
+        """
+        for cell in self._population.cells:
+            cell.persons = [person for person in cell.persons if not
```

Comment:
I am fairly certain it would be faster to filter these lists. I'm worried more generally that this will be really slow however - can we not loop over the population once, and have a method in the person class that updates the relevant cell and household lists that can be called on every relevant person. But storing a list of travelling people should be much more efficient here.